utils/json-utils/src/main/scala/pl/touk/nussknacker/engine/json/encode/BestEffortJsonSchemaEncoder.scala

```diff
@@ -107,6 +108,19 @@ class BestEffortJsonSchemaEncoder(validationMode: ValidationMode) {
   private def error(runtimeType: String, schema: String, fieldName: Option[String]): Invalid[NonEmptyList[String]] =
     Invalid(NonEmptyList.of(s"Not expected type: ${runtimeType} for field${fieldName.map(f => s": '$f'").getOrElse("")} with schema: $schema"))
 
+  private def encodeWithJsonValidation(value: Any, schema: Schema, fieldName: Option[String] = None): EncodeOutput =
+    encode(value, schema, fieldName).andThen { result =>
+      val tryToValid: Try[Json] = Try {
+        val json = JsonSchemaUtils.circeToJson(result)
+        schema.validate(json) //validate can change json object.. e.g. fill default values
+        JsonSchemaUtils.jsonToCirce(json)
+      }
+
+      Validated
+        .fromTry(tryToValid)
+        .leftMap(e => NonEmptyList.of(e.getMessage))
+    }
+
   def encode(value: Any, schema: Schema, fieldName: Option[String] = None): EncodeOutput = {
```

Comment:
this method is public and allows encoding without validation

Reply:
This method is public because we use it at `BestEffortJsonSchemaEncoderTest`. We can't make it private because package of `BestEffortJsonSchemaEncoderTest` is different then `BestEffortJsonSchemaEncoder` but maybe I will change it..  It's intentional to make  `encodeWithJsonValidation` as separate method, the end result will be tested on functional tests. I think there is no need to test whole cases here because we will have to test them again on functional tests.